js/util/util.js

```diff
@@ -41,6 +41,14 @@ exports.wrap = function (n, min, max) {
     return n === max ? n : ((n - min) % d + d) % d + min;
 };
 
+exports.coalesce = function() {
+    for (var i = 0; i < arguments.length; i++) {
+        var arg = arguments[i];
+        if (arg !== null && arg !== undefined)
+            return arg;
```

Comment:
I'd rather not, because It means you can't set a breakpoint on the conditional statement, only on the condition itself. Separate line + no curly braces is my preference, but I'll settle for separate line + always curly braces if there's consensus for that.